cranelift/filetests/src/function_runner.rs

```diff
@@ -66,27 +66,47 @@ impl SingleFunctionCompiler {
     ///  - compile the [Function]
     ///  - compile a `Trampoline` for the [Function]'s signature (or used a cached `Trampoline`;
     ///    this makes it possible to call functions when the signature is not known until runtime.
-    pub fn compile(&mut self, function: Function) -> Result<CompiledFunction, CompilationError> {
+    pub fn compile(self, function: Function) -> Result<CompiledFunction, CompilationError> {
```

Comment:
Changing `SingleFunctionCompiler::compile` from taking `&mut self` to just `self` feels unfortunate, since it forces callers to build a new `TargetIsa` for every function they compile. What would you think of changing `JITBuilder` to take a shared reference instead? Either `&dyn TargetIsa` or `Rc<dyn TargetIsa>`, whichever is easier. In the latter case, `SingleFunctionCompiler` would also take an `Rc` instead of a `Box`.

Reply:
I remember (sorry, this was a while ago) trying to change to `&dyn TargetIsa`, but it brought a bunch of changes due to lifetimes, and I gave up on that path.

I didn't  try `Rc<>` but see the comment down below.